src/Nodes1DProvisioner.cpp

```diff
@@ -373,181 +352,189 @@ namespace blitzdg {
      */
-    const matrix_type & Nodes1DProvisioner::get_Fx() {
+    const matrix_type & Nodes1DProvisioner::get_Fx() const {
         return *Fx;
     }
 
     /**
      * Get the Face-scaling factor (Inverse of Jacobian at Face nodes).
      */
 
-    const matrix_type & Nodes1DProvisioner::get_Fscale() {
+    const matrix_type & Nodes1DProvisioner::get_Fscale() const {
         return *Fscale;
     }
 
     /**
      * Get the index-mask for the face nodes.
      */
-    const index_vector_type & Nodes1DProvisioner::get_Fmask() {
+    const index_vector_type & Nodes1DProvisioner::get_Fmask() const {
         return *Fmask;
     }
 
     /**
      * Get the volume to surface map, 'minus' traces.
      */
-    const index_vector_type & Nodes1DProvisioner::get_vmapM() {
+    const index_vector_type & Nodes1DProvisioner::get_vmapM() const {
         return *vmapM;
     }
 
     /**
      * Get the volume to surface map, 'plus' traces.
      */
-    const index_vector_type & Nodes1DProvisioner::get_vmapP() {
+    const index_vector_type & Nodes1DProvisioner::get_vmapP() const {
         return *vmapP;
     }
 
     /**
      * Get the surface index of the inflow boundary.
      */
-    const index_type Nodes1DProvisioner::get_mapI() {
+    index_type Nodes1DProvisioner::get_mapI() const {
         return mapI;
     }
 
     /**
      * Get the surface index of the outflow boundary.
      */
-    const index_type Nodes1DProvisioner::get_mapO() {
+    index_type Nodes1DProvisioner::get_mapO() const {
         return mapO;
     }
 
     /**
      * Get the volume index of the inflow boundary.
      */
-    const index_type Nodes1DProvisioner::get_vmapI() {
+    index_type Nodes1DProvisioner::get_vmapI() const {
         return vmapI;
     }
 
     /**
      * Get the volume index of the outflow boundary.
      */
-    const index_type Nodes1DProvisioner::get_vmapO() {
+    index_type Nodes1DProvisioner::get_vmapO() const {
         return vmapO;
     }
 
     /**
      * Destructor.
      */
     Nodes1DProvisioner::~Nodes1DProvisioner() {
-        delete rGrid;
-        delete xGrid;
-        delete J;
-        delete rx;
-        delete Lift;
-        delete EToV;
-        delete EToE;
-        delete EToF;
-        delete Fmask;
-        delete Fx;
-        delete Fscale;
-        delete nx;
-        delete vmapM;
-        delete vmapP;
+        delete V; V = nullptr;
+        delete Dr; Dr = nullptr;
+        delete rGrid; rGrid = nullptr;
+        delete xGrid; xGrid = nullptr;
+        delete J; J = nullptr;
+        delete rx; rx = nullptr;
+        delete Lift; Lift = nullptr;
+        delete EToV; EToV = nullptr;
+        delete EToE; EToE = nullptr;
+        delete EToF; EToF = nullptr;
+        delete Fmask; Fmask = nullptr;
+        delete Fx; Fx = nullptr;
+        delete Fscale; Fscale = nullptr;
+        delete nx; nx = nullptr;
+        delete vmapM; vmapM = nullptr;
+        delete vmapP; vmapP = nullptr;
```

Comment:
I always thought the delete would nullify the pointer itself as well, but i guess I shouldn't be surprised that it doesn't...

Reply:
Yah, it would make sense that it would, and depending on the compiler implementation it may do so. Assigning these pointers to nullptr is really just a best practice to be on the safe side.